content/en/blog/2022/frontend-overhaul/index.md

```diff
@@ -0,0 +1,182 @@
+
+Before starting the development process, the frontend application wasn’t
+matching the theme that OpenTelemetry had been using in terms of colors,
+products, and overall user experience. In addition, the demo lacked a real
+frontend (browser side) application as the current implementation was a Go SSR
+application.
+
+![Open Telemetry Demo Old Frontend](old-design.png)
+
+The first task at hand was to bring the demo to the modern age by updating the
+design, color schemes, and user experience. Olly Babiak walked into the fray to
+help us achieve this by creating a modernized version of the application. It
+included an improved way to display the products landing page, an updated
+product details page, a mini cart, and a fully compatible mobile version of the
+application.
+
+![Open Telemetry Demo New Frontend](new-design.png)
+
+Now we had an application design that would match the rest of the OpenTelemetry
+themes and colors and look more like the OpenTelemetry.io website.
+
+## Frontend Application Architecture Overhaul
+
+We worked on an initial proposal that included the following bullet points:
+
+- Framework and tooling (Scaffolding, I/O, styling, UI library)
+- Code Architecture and structure (Directories, coding patterns)
+- Instrumentation
+- Deployment & Distribution
+- Testing (E2E, unit test)
+
+This proposal was presented to the OpenTelemetry Demo SIG duringone of the
+weekly Monday meetings and we were given the green light to move ahead. As part
+of the changes, we decided to use Next.js to not only work as the primary
+frontend application but also to work as an aggregation layer between the
+frontend and the GRPC backend services.
+
+![New Frontend Data Flow](data-flow.png)
+
+As you can see in the Diagram, the application has two major connectivity
+points, one coming from the browser side (REST) to connect to the Next.js
+aggregation layer and the other from the aggregation layer to the backend
+services (GRPC).
+
+## OpenTelemetry Instrumentation
+
+The next big thing we worked on was having a way to instrument both sides of the
+Next.js application. To accommodate this we had to connect the same application
+twice to the same open telemetry collector that was being used by all of the
+other microservices.
+
+For the backend side, a simple solution was designed that involved using the
+[official GRPC exporter](https://www.npmjs.com/package/@opentelemetry/exporter-trace-otlp-grpc)
+in combination with the
+[Nodejs SDK](https://www.npmjs.com/package/@opentelemetry/sdk-node).
+
+You can find the full
+[implementation here](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/frontend/utils/telemetry/Instrumentation.js).
+The basic instrumentation includes auto instrumentation for most of the commonly
+used
+[libraries and tools for Node.js](https://www.npmjs.com/package/@opentelemetry/auto-instrumentations-node).
+But as part of having a better example for users, a manual instrumentation piece
```

Comment:
Rewording:
“As part of providing a better example for users, a manual instrumentation in the form of route middleware was added. This would catch the …”